packages/number-field/src/NumberField.ts

```diff
@@ -474,15 +476,22 @@ export class NumberField extends TextfieldBase {
         this.inputElement.setSelectionRange(nextSelectStart, nextSelectStart);
     }
 
-    private validateInput(value: number): number {
-        const signMultiplier = value < 0 ? -1 : 1; // 'signMultiplier' adjusts 'value' for 'validateInput' and reverts it before returning.
-        value *= signMultiplier;
+    private valueWithLimits(nextValue: number): number {
+        let value = nextValue;
         if (typeof this.min !== 'undefined') {
             value = Math.max(this.min, value);
         }
         if (typeof this.max !== 'undefined') {
             value = Math.min(this.max, value);
         }
+        return value;
+    }
+
+    private validateInput(value: number): number {
+        value = this.valueWithLimits(value);
```

Comment:
Thanks for the deep QE here @spdev3000. It seems this corrects the `-20000` and `-255` input issues you found.

@blunteshwar can you take a look as it seems this stems from your work in https://github.com/adobe/spectrum-web-components/pull/3673/files

Reply:
Looks good to me!
It seems that due to Initially multiplying by the `signMultiplier`  `value = Math.max(this.min, value);` was not able to limit the value.